dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/config_bar_chart_styles.tsx

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useMemo, useCallback } from 'react';
+import { EuiAccordion, EuiSpacer } from '@elastic/eui';
+import { ButtonGroupItem } from './config_button_group';
+import { IConfigPanelOptionSection } from '../../../../../../../../common/types/explorer';
+
+export const ConfigBarChartStyles = ({
+    visualizations,
+    schemas,
+    vizState,
+    handleConfigChange,
+    sectionName,
+    sectionId = 'chartStyles'
+}: any) => {
+    const { data } = visualizations;
+    const { data: vizData = {}, metadata: { fields = [] } = {} } = data?.rawVizData;
+
+    const handleConfigurationChange = useCallback(
+        (stateFieldName) => {
+            return (changes) => {
+                handleConfigChange({
+                    ...vizState,
+                    [stateFieldName]: changes,
+                });
+            };
+        },
+        [handleConfigChange, vizState]
+    );
+
+    /* To update the schema options based on current style mode selection */
+    const currentSchemas = useMemo(() => {
+        if (vizState?.orientation === 'h') {
+            return schemas.filter((schema: IConfigPanelOptionSection) => schema.mapTo !== 'rotateBarLabels');
+        }
+        return schemas;
+    }, [vizState]);
+
+    const dimensions = useMemo(() =>
+        currentSchemas.map((schema: IConfigPanelOptionSection, index: number) => {
+            let params;
+            const DimensionComponent = schema.component || ButtonGroupItem;
+
+            const createDimensionComponent = (dimProps) => (
+                <>
+                    <DimensionComponent key={`viz-series-${index}`} {...dimProps} />
+                    <EuiSpacer size="s" />
+                </>
+            )
+            if (schema.eleType === 'buttons') {
+                params = {
+                    title: schema.name,
+                    legend: schema.name,
+                    groupOptions: schema?.props?.options.map((btn: { name: string }) => ({ ...btn, label: btn.name })),
+                    idSelected: vizState[schema.mapTo] || schema?.props?.defaultSelections[0]?.id,
+                    handleButtonChange: handleConfigurationChange(schema.mapTo),
+                    vizState,
+                    ...schema.props,
+                };
+                return createDimensionComponent(params);
+            }
+
+            if (schema.eleType === 'slider') {
```

Comment:
Suggestion: Rather than using two separate if statements we can use 'else if' to improve efficiency.

Reply:
Hi @kavithacm

As in the upcoming feature implementation, there will be more UI on config. So we are creating this separate if block statement to form respective params for similar elements. And as we are returning in every if statement, it will behave as if-else.